### PR TITLE
Frontend Changes for Forgot Password

### PR DESCRIPTION
**Pull request by su-per-man**
The front end is ready for the password reset functionality.

### CODE REVIEW

File: front/src/components/header/subcomponents/LoginModal.tsx [hunk 1]
```diff
@@ -40,6 +41,7 @@ type Props = OwnProps & DispatchProps;
 type FormData = {
   email: string;
   password: string;
+  otp: string;
```
Comment (trybick) [inline front/src/components/header/subcomponents/LoginModal.tsx]: Can we change all instances to `oneTimeCode`? Including the API endpoint please {API.TV_MINDER}/oneTimeCodeVerification

File: front/src/components/header/subcomponents/LoginModal.tsx [hunk 2]
```diff
@@ -51,6 +53,9 @@ const formSchema = {
   password: {
     required: 'Password is required',
   },
+  otp: {
+    required: 'OTP is required',
```
Comment (trybick) [inline front/src/components/header/subcomponents/LoginModal.tsx]: 'One time code is required'

File: front/src/components/header/subcomponents/LoginModal.tsx [hunk 3]
```diff
@@ -92,56 +116,136 @@ const LoginModal = ({ disclosureProps, setIsLoggedIn, unregisteredClearFollowedS
...
+          status: 'success',
+          isClosable: true,
+        });
+      })
+      .catch(err => {
+        handleErrors(err);
+        setIsLoading(false);
+        setError('login', 'generic', 'The email is not registered');
+      });
+  };
```
Comment (trybick) [inline front/src/components/header/subcomponents/LoginModal.tsx:138]: nit - add space between each function

File: front/src/components/header/subcomponents/LoginModal.tsx [hunk 4]
```diff
@@ -92,56 +116,136 @@ const LoginModal = ({ disclosureProps, setIsLoggedIn, unregisteredClearFollowedS
...
+              handleFormClose();
             }}
           />
-
           <ModalBody pb={6}>
             <FormControl isInvalid={Boolean(errors.email)}>
               <FormLabel htmlFor="email">Email</FormLabel>
-              <Input name="email" placeholder="Email" ref={register(formSchema.email)} autoFocus />
+              <Input
+                isDisabled={formOption === 0 || formOption === 1 ? false : true}
```
Comment (trybick) [inline front/src/components/header/subcomponents/LoginModal.tsx]: could this work?

`isDisabled={formOption === 2 || formOption === 3}`

File: front/src/components/header/subcomponents/LoginModal.tsx [hunk 5]
```diff
@@ -92,56 +116,136 @@ const LoginModal = ({ disclosureProps, setIsLoggedIn, unregisteredClearFollowedS
...
+              <FormControl isInvalid={Boolean(errors.password)} mt={4}>
+                <FormLabel>Enter Verification Code</FormLabel>
+                <Input name="otp" placeholder="OTP" ref={register(formSchema.otp)} />
+                <FormErrorMessage>{errors.otp?.message}</FormErrorMessage>
+              </FormControl>
+            )}
             <FormControl isInvalid={Boolean(errors.login)} mt={4}>
               <FormErrorMessage>{errors.login?.message}</FormErrorMessage>
             </FormControl>
+            <Link color="teal.500" onClick={() => setformOption((formOption + 1) % 2)}>
```
Comment (trybick) [inline front/src/components/header/subcomponents/LoginModal.tsx]: I think the `(formOption + 1) % 2)` could be confusing - pls add a comment about what it's doing

File: front/src/components/header/subcomponents/LoginModal.tsx [hunk 6]
```diff
@@ -92,56 +116,136 @@ const LoginModal = ({ disclosureProps, setIsLoggedIn, unregisteredClearFollowedS
         setError('login', 'generic', 'Invalid login. Please try again.');
         setValue('password', '');
       });
-  });
+  };
+  const processOtp = (email: string) => {
+    axios
+      .post(`${API.TV_MINDER}/processotp`, email)
+      .then(res => {
```
Comment (trybick) [inline front/src/components/header/subcomponents/LoginModal.tsx]: I think `res` can be removed and just have `()` (same for functions below)

File: front/src/components/header/subcomponents/LoginModal.tsx [hunk 7]
```diff
@@ -92,56 +116,136 @@ const LoginModal = ({ disclosureProps, setIsLoggedIn, unregisteredClearFollowedS
...
+        setIsLoading(false);
+        setformOption(2);
+        toast({
+          title: 'Password Reset!',
+          description: 'A password reset mail has been sent',
```
Comment (trybick) [inline front/src/components/header/subcomponents/LoginModal.tsx]: Update to 'A one-time code has been sent to your email address.'

File: front/src/components/header/subcomponents/LoginModal.tsx [hunk 8]
```diff
@@ -66,19 +71,38 @@ const LoginModal = ({ disclosureProps, setIsLoggedIn, unregisteredClearFollowedS
   const [passwordVisible, setPasswordVisible] = React.useState(false);
   const togglePasswordVisible = () => setPasswordVisible(!passwordVisible);
 
-  const onSubmit = handleSubmit(({ email, password }) => {
+  //Forgot Password
+  const [formOption, setformOption] = React.useState(0);
```
Comment (trybick) [inline front/src/components/header/subcomponents/LoginModal.tsx]: update to `setFormOption`

**Review by trybick — COMMENTED** [review]
Looks really good! I think once these are addressed this will be great to merge.

**Comment by su-per-man** [comment]
Sure. Will update soon.

File: front/src/components/header/subcomponents/LoginModal.tsx (hunk 3 above)
Comment (su-per-man) [reply front/src/components/header/subcomponents/LoginModal.tsx:138]: Sorry, didn't get it.

**Comment by trybick** [comment]
Awesome! thanks @suman98kumar

**Pull request merged**